Obsidian/Net/Packets/Play/Serverbound/PlayerBlockPlacement.cs

```diff
@@ -0,0 +1,362 @@
+                    await player.client.QueuePacketAsync(new BlockAction
+                    {
+                        Location = location,
+                        ActionId = 1,
+                        ActionParam = 1,
+                        BlockType = interactedBlock.Id
+                    });
+                    await player.SendSoundAsync(Sounds.BlockChestOpen, location.SoundPosition, SoundCategory.Blocks);
+
+                    var invUuid = Guid.NewGuid();
+
+                    var blockMeta = new BlockMetaBuilder().WithInventoryId(invUuid).Build();
+
+                    server.World.SetBlockMeta(location, blockMeta);
+
+                    server.CachedWindows.TryAdd(invUuid, inventory);
+
+                    player.OpenedInventory = inventory;
+                }
+                else if (type == Materials.EnderChest)
+                {
+                    var enderChest = new Inventory(player.Uuid)
+                    {
+                        Title = "Ender Chest",
+                        Type = InventoryType.Generic,
+                        Size = 9 * 3,
+                    };
+
+                    var invUuid = Guid.NewGuid();
+
+                    var blockMeta = new BlockMetaBuilder().WithInventoryId(invUuid).Build();
+
+                    server.World.SetBlockMeta(location, blockMeta);
+
+                    server.CachedWindows.TryAdd(invUuid, enderChest);
+
+                    player.OpenedInventory = enderChest;
+
+                    await player.OpenInventoryAsync(enderChest);
+                    await player.client.QueuePacketAsync(new BlockAction
+                    {
+                        Location = location,
+                        ActionId = 1,
+                        ActionParam = 1,
+                        BlockType = interactedBlock.Id
+                    });
+                    await player.SendSoundAsync(Sounds.BlockEnderChestOpen, location.SoundPosition, SoundCategory.Blocks);
+                }
+                else if (type == Materials.CraftingTable)
+                {
+                    var crafting = new Inventory
+                    {
+                        Title = "Crafting Table",
+                        Type = InventoryType.Crafting,
+                        Id = maxId,
+                        Size = 2 * 5,
+                        BlockPosition = location
+                    };
+
+                    player.OpenedInventory = crafting;
+
+                    await player.OpenInventoryAsync(crafting);
+                }
+                else if (type == Materials.Furnace || type == Materials.BlastFurnace || type == Materials.Smoker)
+                {
+                    InventoryType actualType = type == Materials.Furnace ? InventoryType.Furnace :
+                        type == Materials.BlastFurnace ? InventoryType.BlastFurnace : InventoryType.Smoker;
+
+                    var furnace = new Inventory
+                    {
+                        Title = "",
```

Comment:
These empty strings are a bit too explicit, maybe make those the default value for inventories? (also consider using `String.Empty` instead of `""`)